app/services/Payment.js

```diff
@@ -16,19 +17,47 @@ class Payment extends Service {
         return this.fetchJson(url, fetchOptions);
     }
 
+    getCasePayments(data) {
+        this.log('Getting all payments from case');
+        const url = `${this.endpoint}?service_name=Probate&ccd_case_number=${data.caseId}`;
+        const headers = {
+            'Content-Type': 'application/json',
+            'Authorization': data.authToken,
+            'ServiceAuthorization': data.serviceAuthToken
+        };
+        const fetchOptions = this.fetchOptions(data, 'GET', headers);
+        return this.fetchJson(url, fetchOptions);
+    }
+
     post(data, hostname) {
         this.log('Post payment');
         const url = this.endpoint;
         const headers = {
             'Content-Type': 'application/json',
             'Authorization': data.authToken,
             'ServiceAuthorization': data.serviceAuthToken,
-            'return-url': this.formatUrl.format(hostname, '/payment-status')
+            'return-url': this.formatUrl.format(hostname, config.services.payment.paths.returnUrlPath)
         };
         const body = paymentData.createPaymentData(data);
         const fetchOptions = this.fetchOptions(body, 'POST', headers);
         return [this.fetchJson(url, fetchOptions), body.reference];
     }
+
+    identifySuccessfulOrInitiatedPayment(casePayments) {
+        for (const payment of casePayments.payments) {
+            if (payment.status === 'Success') {
+                this.log(`Found successful payment: ${payment.payment_reference}`);
+                return payment;
+            }
+        }
+        for (const payment of casePayments.payments) {
```

Comment:
This is because when checking the payments for a case, if I find a Success then I can ignore any Initiated states. I can only do this if run through the entire array of payments first.